src/fastoad/model_base/openmdao/group.py

```diff
@@ -0,0 +1,175 @@
+            desc="If True, solvers are added to the group. The solver classes are decided "
+            'by options "linear_solver" and "nonlinear_solver". '
+            "If False, no solver is added and other solver-related options have no effect.",
+        )
+        self.options.declare(
+            "use_inner_solver",
+            types=bool,
+            default=self.use_solvers_by_default,
+            deprecation=(
+                '"use_inner_solver" option is deprecated. Please use "use_solvers" option',
+                "use_solvers",
+            ),
+        )
+        self.options.declare(
+            "linear_solver",
+            types=(bool, str),
+            default=self.default_linear_solver,
+            desc="If a string is given like `om.LinearBlockGS`, it will be used "
+            "(convention `import openmdao.api as om' is assumed`. "
+            "If `False` is given, no linear solver is added (i.e. om.LinearRunOnce is used). "
+            'Ignored if option "use_inner_solver" is False',
+            check_valid=_forbid_true_value,
+        )
+        self.options.declare(
+            "nonlinear_solver",
+            types=(bool, str),
+            default=self.default_nonlinear_solver,
+            desc="If a string is given like `om.NewtonSolver`, it will be used "
+            "(convention `import openmdao.api as om' is assumed`. "
+            "If `False` is given, no non-linear solver is added (i.e. om.NonlinearRunOnce is used)."
+            'Ignored if option "use_inner_solver" is False',
+            check_valid=_forbid_true_value,
+        )
+        self.options.declare(
+            "linear_options",
+            types=dict,
+            default=self.default_solver_options["linear_options"],
+            desc="Options for linear solver. This dict will be updated with provided dict, "
+            "meaning that options that are not in provided dict will keep the default value. "
+            'Ignored if "use_inner_solver" is False.',
+        )
+        self.options.declare(
+            "nonlinear_options",
+            types=dict,
+            default=self.default_solver_options["nonlinear_options"],
+            desc="Options for non-linear solver. This dict will be updated with provided dict, "
+            "meaning that options that are not in provided dict will keep the default value. "
+            'Ignored if "use_inner_solver" is False.',
+        )
+
+    def setup(self):
+        super().setup()
+
+        if self.options["use_solvers"]:
+            linear_solver_class = self._get_solver_class("linear_solver")
+            nonlinear_solver_class = self._get_solver_class("nonlinear_solver")
+
+            linear_options = self._get_solver_options("linear_options")
+            nonlinear_options = self._get_solver_options("nonlinear_options")
+
+            if linear_solver_class:
+                self.linear_solver = linear_solver_class(**linear_options)
```

Comment:
Are the solver instantiated here ? If yes, I can't really see where we transition form a solver name in the form of a string and the actual  solver

Reply:
> Are the solver instantiated here ? If yes, I can't really see where we transition form a solver name in the form of a string and the actual solver

It happens in line 148. I used `eval`, at first, but retrieving them from `om.__dict__` is less prone to security problems.